be/src/util/metrics.cpp

```diff
@@ -139,7 +139,7 @@ void MetricCollector::get_metrics(std::vector<Metric*>* metrics) {
 
 MetricRegistry::~MetricRegistry() noexcept {
     {
-        std::unique_lock lock(_mutex);
+        std::unique_lock lock(_collector_mutex);
```

Comment:
whether need to clear the hook explicitly under hook_mutex?

Reply:
Done